tasties_app/models.py

```diff
@@ -46,6 +55,65 @@ class UnitChoices(models.TextChoices, Enum):
     )
     description = models.CharField(max_length=64, validators=[MinLengthValidator(1)])
 
+    def __str__(self):
+        return str(self.amount) + " " + self.measurement_unit + " " + self.description
+
+    @staticmethod
+    def is_enum_value(value):
+        """
+        Checks if the given value is a valid UnitChoices enum value.
+        """
+        for unit in Ingredient.UnitChoices:
+            if unit == value:
+                return True
+        return False
+
+    @staticmethod
+    def add(recipe, amount, measurement_unit, description):
+        """
+        Adds new ingredient instance to our database.
+            recipe = the recipe that this ingredient is related to
+            amount = the amount of the recipe
+            measurement_unit = measurement_unit
+            description = describes the ingredient
+        """
+        try:
+            if amount < 0.0001:
+                raise ValueError("Invalid amount value")
+            if not Ingredient.is_enum_value(measurement_unit):
+                raise ValueError("Invalid measurement_unit value")
+            if len(description) < 1:
+                raise ValueError("Empty Description")
```

Comment:
Why do you need to do this validation if you have the validators in the class attributes?

Reply:
Because the validators in the class won't work when I edit the attributes from the edit method. I can easily insert a negative value to the amount attribute, for example, and it will be saved to the database.